model/histogram/float_histogram.go

```diff
@@ -130,6 +130,60 @@ func (h *FloatHistogram) String() string {
 	return sb.String()
 }
 
+// PromQlExpression returns a PromQL expression that evaluates to this histogram when used in promtool.
+func (h *FloatHistogram) PromQlExpression() string {
+	trim := func(s string) string {
+		for {
+			found := false
+			s, found = strings.CutSuffix(s, "0")
+			if !found {
+				break
+			}
+		}
+		return strings.TrimSuffix(s, ".")
+	}
+	var res []string
+	if h.Schema != 0 {
+		res = append(res, fmt.Sprintf("schema:%d", h.Schema))
+	}
+	if h.Count != 0 {
+		res = append(res, trim(fmt.Sprintf("count:%f", h.Count)))
+	}
+	if h.Sum != 0 {
+		res = append(res, trim(fmt.Sprintf("sum:%f", h.Sum)))
+	}
+	if h.ZeroCount != 0 {
+		res = append(res, trim(fmt.Sprintf("z_bucket:%f", h.ZeroCount)))
+	}
+	if h.ZeroThreshold != 0 {
+		res = append(res, trim(fmt.Sprintf("z_bucket_w:%f", h.ZeroThreshold)))
+	}
+
+	addBuckets := func(kind, bucketsKey, offsetKey string, buckets []float64, spans []Span) []string {
+		if len(spans) > 1 {
+			panic(fmt.Sprintf("histogram with multiple %s spans not supported", kind))
```

Comment:
Happy to fix it - do you have an idea?

Reply:
You could fill the gaps between spans with empty buckets.

Reply:
good idea - done.

Reply:
With your explanation (given elsewhere) I understand now how the problem got solved. See above, please add a comment to `m.Compact(math.MaxInt)` so that the reader understands this little trick.